src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -63,12 +62,12 @@ public static void Run(AddSourceArgs args, Func<ILogger> getLogger)
 
             var newPackageSource = new Configuration.PackageSource(args.Source, args.Name);
 
-            if (newPackageSource.IsHttp && !newPackageSource.IsHttps && !newPackageSource.AllowInsecureConnections)
+            if (newPackageSource.IsHttp && !newPackageSource.IsHttps)
```

Comment:
When we're adding, can the error message be more empowering and tell the user what they need to pass as an extra option?

This command is special enough where the generic error message can be improved.

idk if that conflicts with the other motivation behind deprecating http, so curious what others think

Reply:
Currently, the error message they get is `You are running the '{0}' operation with an 'HTTP' source: {1}. NuGet requires HTTPS sources. Please refer to https://aka.ms/nuget-https-everywhere.` I removed the information about editing the config file as that would not make sense when trying to add a new source. 

From my understanding, I do not think we provide users with the option to add an http source using this command. I was considering of adding a new option that would allow users to add an http source. May be `-allowInsecureConnections`, similar to our config settings.

Reply:
Let's have a discussion about this, but I think the new option might be necessary.

As this is, we'd effectively block people from adding http sources period, with no automation based workaround available.